can/notifier.py

```diff
@@ -13,57 +13,62 @@
 
 class Notifier(object):
 
-    def __init__(self, bus, listeners, timeout=None):
-        """Manages the distribution of **Messages** from a given bus to a
+    def __init__(self, bus, listeners, timeout=1):
+        """Manages the distribution of **Messages** from a given bus/buses to a
         list of listeners.
 
-        :param bus: The :ref:`bus` to listen too.
-        :param listeners: An iterable of :class:`~can.Listener`s
-        :param timeout: An optional maximum number of seconds to wait for any message.
+        :param bus: The :ref:`bus` or list of buses to listen to.
+        :param list listeners: An iterable of :class:`~can.Listener`s
+        :param float timeout: An optional maximum number of seconds to wait for any message.
         """
         self.listeners = listeners
         self.bus = bus
         self.timeout = timeout
 
-        # exception raised in thread
+        #: Exception raised in thread
         self.exception = None
 
-        self._running = threading.Event()
-        self._running.set()
+        self._running = True
+        self._lock = threading.Lock()
 
-        self._reader = threading.Thread(target=self._rx_thread,
-                                        name='can.notifier for bus "{}"'.format(self.bus.channel_info))
-        self._reader.daemon = True
-        self._reader.start()
+        self._readers = []
+        buses = self.bus if isinstance(self.bus, list) else [self.bus]
+        for bus in buses:
+            reader = threading.Thread(target=self._rx_thread, args=(bus,),
+                                      name='can.notifier for bus "{}"'.format(bus.channel_info))
+            reader.daemon = True
+            reader.start()
+            self._readers.append(reader)
 
     def stop(self):
         """Stop notifying Listeners when new :class:`~can.Message` objects arrive
         and call :meth:`~can.Listener.stop` on each Listener.
         """
-        self._running.clear()
-        if self.timeout is not None:
-            self._reader.join(self.timeout + 0.1)
-
-    def _rx_thread(self):
+        self._running = False
+        timeout = self.timeout + 0.1 if self.timeout is not None else 5
+        for reader in self._readers:
+            reader.join(timeout)
```

Comment:
maybe add a timeout parameter to the method. Additionally, it might be surprising that the timeout counts for each bus. So for five channels and a timeout of one seconds, this method could block 5 seconds although many would expect it to block 1 second.